src/ripple/conditions/impl/Condition.cpp

```diff
@@ -29,160 +29,173 @@
+// sub-field.  Some types, such as the compound condition
+// types, define additional sub-fields that are required to
+// convey essential properties of the cryptocondition (such
+// as the sub-types used by sub-conditions in the case of
+// the compound types).
+//
+// Conditions are encoded as follows:
+//
+//    Condition ::= CHOICE {
+//      preimageSha256   [0] SimpleSha256Condition,
+//      prefixSha256     [1] CompoundSha256Condition,
+//      thresholdSha256  [2] CompoundSha256Condition,
+//      rsaSha256        [3] SimpleSha256Condition,
+//      ed25519Sha256    [4] SimpleSha256Condition
+//    }
+//
+//    SimpleSha256Condition ::= SEQUENCE {
+//      fingerprint          OCTET STRING (SIZE(32)),
+//      cost                 INTEGER (0..4294967295)
+//    }
+//
+//    CompoundSha256Condition ::= SEQUENCE {
+//      fingerprint          OCTET STRING (SIZE(32)),
+//      cost                 INTEGER (0..4294967295),
+//      subtypes             ConditionTypes
+//    }
+//
+//    ConditionTypes ::= BIT STRING {
+//      preImageSha256   (0),
+//      prefixSha256     (1),
+//      thresholdSha256  (2),
+//      rsaSha256        (3),
+//      ed25519Sha256    (4)
+//    }
+
+std::unique_ptr<Condition>
+loadSimpleSha256(Type type, Slice s, std::error_code& ec)
 {
-    static boost::regex const re_current (
-            "^"                            // start of line
-            "cc:"                          // 'cc' for cryptocondition
-            "([1-9a-f][0-9a-f]{0,3}|0):"   // type (hexadecimal)
-            "([1-9a-f][0-9a-f]{0,15}):"    // feature bitmask (hexadecimal)
-            "([a-zA-Z0-9_-]{0,86}):"       // fingerprint (base64url)
-            "([1-9][0-9]{0,17}|0)"         // fulfillment length (decimal)
-            "$"                            // end of line
-        , boost::regex_constants::optimize
-    );
-
-    boost::smatch match;
-
-    if (!boost::regex_match (s, match, re_current))
-        return boost::none;
-
-    try
-    {
-        Condition c;
-
-        c.type = parse_hexadecimal<std::uint16_t> (match[1]);
-
-        if (!isCondition (c.type))
-            return boost::none;
-
-        c.featureBitmask = parse_hexadecimal<std::uint32_t>(match[2]);
-        c.maxFulfillmentLength = parse_decimal<std::uint16_t>(match[4]);
-
-        if (c.maxFulfillmentLength > maxSupportedFulfillmentLength)
-            return boost::none;
+    using namespace der;
 
-        // TODO: Avoid copying by decoding directly
-        // into the condition's buffer
-        auto fingerprint = base64url_decode(match[3]);
+    auto p = parsePreamble(s, ec);
 
-        if (fingerprint.size() != c.fingerprint.size())
-            return boost::none;
+    if (ec)
+        return {};
 
-        std::memcpy(
-            c.fingerprint.data(),
-            fingerprint.data(),
-            fingerprint.size());
-
-        return c;
-    }
-    catch (std::exception const&)
+    if (p.tag != 0)
```

Comment:
This doesn't look right. The serialization should be: `SEQUENCE {OCTET STRING(SIZE(32)), INTEGER(0..4294967295))`.

So the tags should be `16` then `4` then `2`. We should also check that the octet string group is exactly 32 bytes. We should also check the preamble type matches our expected type (i.e. `universal`)